### PR TITLE
адаптивная верстка и декоративные элементы

### PR DESCRIPTION
**Pull request by hinhilla**

### CODE REVIEW

File: source/less/blocks/page.less [hunk 1]
```diff
@@ -0,0 +1,15 @@
+.page {
+box-sizing: border-box;
+
+}
+.page__body {
```
Comment (PetrischakMaxim) [inline source/less/blocks/page.less:5]: размер шрифта, цвет и фоновый цвета лучше тоже добавить.
Так же можно добавить минимальную высоту 578px iphone 5

File: source/less/blocks/page-footer.less [hunk 2]
```diff
@@ -0,0 +1,40 @@
+.page-footer {
+  display: flex;
+      justify-content: center;
+
+  background-color: @base-blue;
+  padding-top: 22px;
+  padding-bottom: 22px;
+  padding-left: 40px;
```
Comment (PetrischakMaxim) [inline source/less/blocks/page-footer.less]: padding-left: 40px;
 padding-right: 40px;
скорее всего паддинги и центрование лучше вынести в отдельный класс центровщик.
Попробуй сделать универсальный центровщик для всех крупных блоков.
вот пару ссылок:
- [1](https://coderoad.ru/28047651/%D0%97%D0%B0%D1%87%D0%B5%D0%BC-%D0%BE%D0%B1%D0%BE%D1%80%D0%B0%D1%87%D0%B8%D0%B2%D0%B0%D1%82%D1%8C-div-class-container-%D0%B2%D0%BE%D0%BA%D1%80%D1%83%D0%B3-%D0%BD%D0%B0%D0%B2%D0%B8%D0%B3%D0%B0%D1%86%D0%B8%D0%BE%D0%BD%D0%BD%D1%8B%D1%85-%D1%8D%D0%BB%D0%B5%D0%BC%D0%B5%D0%BD%D1%82%D0%BE%D0%B2)
- [2](https://yoksel.github.io/bad-practices/containers/)
- [3](https://css-live.ru/articles/luchshij-sposob-realizovat-obyortki-v-css.html)

File: source/less/blocks/page-header.less [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+  position: relative;
+background-color: @base-blue;
+height: 71px;
+@media (min-width: @tablet) {
+height: 95px;
+}
+
+}
+.page-header__wrapper {
+position: absolute;
```
Comment (PetrischakMaxim) [inline source/less/blocks/page-header.less]: думаю немного все сложно сделано. Давай я скину пример как можно сделать. 
https://petrischakmaxim.github.io/menu/
Посмотри его если что непонятно - пиши.

File: source/less/blocks/page.less [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+min-width: 320px;
+// position: relative;
+margin: auto;
+font-family: "Lato", arial, sans-serif;
```
Comment (PetrischakMaxim) [inline source/less/blocks/page.less]: Есть хорошая практика, вторым шрифтом ставить похожий или аналог шрифта главного, тут просто можно загуглить аналог) И шрифты именуй все одинаково.
font-family: "Lato", "Roboto" , "Arial", sans-serif;

File: source/less/blocks/portfolio.less [hunk 5]
```diff
@@ -0,0 +1,96 @@
+.portfolio {
+background-color: @base-blue;
+padding-right: 30px;
+padding-top: 42px;
+padding-bottom: 42px;
+padding-left: 30px;
+}
+.portfolio__title {
+.header;
```
Comment (PetrischakMaxim) [inline source/less/blocks/portfolio.less]: .header; точно тут надо?

Comment (PetrischakMaxim) [reply source/less/blocks/portfolio.less]: такой подход ухудшает читаемость, если много где такой код повторяеться - просто выноси его в класс и подставляй елементу.

File: source/less/common/mixin.less [hunk 6]
```diff
@@ -0,0 +1,33 @@
+.header {
```
Comment (PetrischakMaxim) [inline source/less/common/mixin.less]: миксин немного отличаються по синтаксису) 
Давай договоримся, ._mixin() вот такой будет вид для миксина

File: source/less/common/mixin.less [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+margin-top: 0;
+margin-bottom: 0;
+}
+.list {
+list-style: none;
+padding-left: 0;
+margin-top: 0;
+margin-bottom: 0;
+}
+// .center {
```
Comment (PetrischakMaxim) [inline source/less/common/mixin.less]: хорошая идея)

File: source/js/script.js [hunk 8]
```diff
@@ -1 +1,18 @@
-console.log('hi')
+const buttonNavigation = document.querySelector('.page-header__navigation-button');
+console.log(buttonNavigation);
+const navigation = document.querySelector('.page-header__navigation-list');
+console.log(navigation);
+const modal = document.querySelector('.page-header__wrapper');
+
+const toToggleBlockByButton = (button, blockNavigation, blockWrapper) => {
+  button.classList.toggle('.page-header__navigation-button--active');
+  blockNavigation.classList.toggle('.page-header__navigation-list--opened');
```
Comment (PetrischakMaxim) [inline source/js/script.js]: в методе toggle имя класса нужно писать без точки.
И будет все работать.
И посмотри пример с меню который кидал ссылкой.

**Review by PetrischakMaxim — COMMENTED** [review]
- Burger-menu-icon.svg лучше именовать графику с маленькой буквы
-

File: source/index.html [hunk 9]
```diff
@@ -1,126 +1,314 @@
...
   <meta name="viewport" content="width=device-width, initial-scale=1.0">
-  <title>Document</title>
+  <title>Singolo</title>
   <link rel="stylesheet" href="./css/style.min.css">
+  <link rel="preconnect" href="https://fonts.googleapis.com">
+  <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
 
+
+
+  <svg style="display: none;">
```
Comment (PetrischakMaxim) [inline source/index.html]: спрайт можно в конец страницы чтоб читаемость была проще

Comment (PetrischakMaxim) [reply source/index.html]: и не в head а перед закрывающим тегом body

File: source/index.html [hunk 10]
```diff
@@ -1,126 +1,314 @@
...
+        fill="#666D89" />
+    </symbol>
+  </svg>
```
Comment (PetrischakMaxim) [inline source/index.html]: По спрайтам 1 прикол можно кстати сделать fill="currentColor"
https://codepen.io/jjsoini/pen/oLJpAv 
Это для того чтобы легче взаимодействовать с цветами при наведении

File: source/index.html [hunk 11]
```diff
@@ -1,126 +1,314 @@
...
-    </nav>
-    <h1 class="page-header__logo">
-      <span class="page-header__logo-name">singolo</span>
-      <span class="page-header__logo-symbol">*</span>
-    </h1>
+<body class="page__body">
+  <header class="page__header page-header">
+    <div class="center--header">
+      <div class="page-header__inner">
+        <button class="page-header__navigation-button">
```
Comment (PetrischakMaxim) [inline source/index.html]: А попробуй положить кнопку во внутрь навигации, ведь это ее часть.

File: source/index.html [hunk 12]
```diff
@@ -1,126 +1,314 @@
...
+              <a href="#portfolio" class="page-header__navigation-link">portfolio</a>
+            </li>
+            <li class="page-header__navigation-item">
+              <svg class="page-header__dot" width="2" height="2">
+                <use xlink:href=" #dot"></use>
+              </svg>
+              <a href="#about" class="page-header__navigation-link">about</a>
+            </li>
+            <li class="page-header__navigation-item">
+              <svg class="page-header__dot" width="2" height="2">
```
Comment (PetrischakMaxim) [inline source/index.html]: В целом эту точку можно было сделать через обычный marker для <li>

File: source/index.html [hunk 13]
```diff
@@ -1,126 +1,314 @@
...
-    <section class="page-main__portfolio portfolio">
-      <h2 class="portfolio__title">Portfolio</h2>
-      <p class="portfolio__paragraph">Duis mollis, est non commodo luctus, nisi erat porttitor ligula,
-        eget lacinia odio sem nec elit.</p>
-      <nav class="portfolio__navigation">
-        <ul class="portfolio__navigation-list">
-          <li class="portfolio__navigation-item"><button class="portfolio__item-button">All</button></li>
-          <li class="portfolio__navigation-item"><button class="portfolio__item-button">Web Design</button>
+    <div class="page-main__slider slider">
+      <button class="slider__button slider__button-left">
```
Comment (PetrischakMaxim) [inline source/index.html]: Немного структура не такая, кнопки можно сгрупировать в обертку, а сами слайды закинуть в список, ведь это слайдер тут будет больше чем 1 слайд.

File: source/index.html [hunk 14]
```diff
@@ -1,126 +1,314 @@
...
+            <li class="portfolio__navigation-item"><button class="portfolio__item-button">Graphic
+                Design</button></li>
+            <li class="portfolio__navigation-item"><button class="portfolio__item-button">Artwork</button></li>
+          </ul>
+        </nav>
+        <ul class="portfolio__list">
+          <li class="portfolio__item"><img src="img/portfolio-1.jpg" alt="Изображение 1" width="128" height=" 108.5"
+              class="portfolio__image"></li>
+          <li class="portfolio__item"><img src="img/portfolio-2.jpg" alt="Изображение 2" class="portfolio__image"
+              width=" 128" height=" 108.5"></li>
```
Comment (PetrischakMaxim) [inline source/index.html]: округляй без дробного числа 109

Comment (PetrischakMaxim) [reply source/index.html]: И тут наверное еще можно ссылкой обернуть, в теории при нажатии можно будет посмотреть работу.

File: source/index.html [hunk 15]
```diff
@@ -1,126 +1,314 @@
...
-    <p class="page-footer__copyright">© Copyright 2013</p>
-    <nav class="page-footer__navigation">
-      <ul class="page-footer__navigation-list">
-        <li class="page-footer__navigation-item">facebook</li>
-        <li class="page-footer__navigation-item">google</li>
-        <li class="page-footer__navigation-item">twitter</li>
-        <li class="page-footer__navigation-item">instagram</li>
-      </ul>
-    </nav>
+    <div class="center">
```
Comment (PetrischakMaxim) [inline source/index.html:208]: Задай центровщику максимальную ширину чтоб контент не расползался при изменении масштаба.
https://prnt.sc/1s783rk

File: source/index.html [hunk 16]
```diff
@@ -1,126 +1,314 @@
...
+              <a href="#" class="page-footer__navigation-link">
+                <span class="visually-hidden">facebook</span>
+                <svg class=" " width="8" height="14">
+                  <use xlink:href=" #facebook"></use>
+                </svg>
+              </a>
+
+            </li>
+            <li class="page-footer__navigation-item">
+              <a href="#" class="page-footer__navigation-link">
```
Comment (PetrischakMaxim) [inline source/index.html:226]: задай ссылкам title с адресом соц сети, сделаем иконку доступнее)

File: source/js/script.js [hunk 17]
```diff
@@ -1 +1,8 @@
-console.log('hi')
+const buttonNavigation = document.querySelector('.page-header__navigation-button');
+const body = document.querySelector('.page__body');
+
+
+const toToggleModalByButton = (body) => body.classList.toggle('show-modal');
```
Comment (PetrischakMaxim) [inline source/js/script.js]: const toToggleModalByButton = (element = bodyElement ) => element.classList.toggle('show-modal');
И тут сократим тогда)
buttonNavigation.addEventListener('click', toToggleModalByButton);

File: source/js/script.js [hunk 18]
```diff
@@ -1 +1,8 @@
-console.log('hi')
+const buttonNavigation = document.querySelector('.page-header__navigation-button');
```
Comment (PetrischakMaxim) [inline source/js/script.js:1]: buttonNavigationElement

File: source/js/script.js [hunk 19]
```diff
@@ -1 +1,8 @@
...
+const body = document.querySelector('.page__body');
```
Comment (PetrischakMaxim) [inline source/js/script.js:2]: bodyElement

File: source/less/blocks/header.less [hunk 20]
```diff
@@ -0,0 +1,9 @@
+
+.header {
```
Comment (PetrischakMaxim) [inline source/less/blocks/header.less:2]: Форматируй сразу код, чтоб было аккуратно.
Вот попробуй https://www.digitalocean.com/community/tutorials/how-to-format-code-with-prettier-in-visual-studio-code-ru

File: source/less/blocks/page-footer.less [hunk 21]
```diff
@@ -0,0 +1,41 @@
+.page-footer {
+
+background-color: @base-blue;
+padding-top: 22px;
+padding-bottom: 22px;
```
Comment (PetrischakMaxim) [inline source/less/blocks/page-footer.less]: padding: 22px 0;
Где можно - сокращай)

File: source/less/blocks/page-header.less [hunk 22]
```diff
@@ -0,0 +1,178 @@
...
+display: flex;
+padding-top: 23px;
+padding-bottom: 23px;
+@media (min-width: @tablet) {
+justify-content: space-between;
+padding-top: 29px;
+padding-bottom: 30px;
+}
+}
+// .page-header__wrapper {
```
Comment (PetrischakMaxim) [inline source/less/blocks/page-header.less]: Что лишнее убирай лучше сразу.

File: source/less/blocks/text.less [hunk 23]
```diff
@@ -0,0 +1,7 @@
+.text {
+color: @section-descriptions-grey;
+font-weight: 300;
+font-size: 18px;
+line-height: 30px;
```
Comment (PetrischakMaxim) [inline source/less/blocks/text.less]: `    font-size: 18px;
    font-weight: 300;
    line-height: 30px;` Все это наследуеться от body , можно не писать

File: source/less/common/center.less [hunk 24]
```diff
@@ -0,0 +1,18 @@
...
+padding-left: 30px;
+padding-right: 30px;
+margin-left: auto;
+margin-right: auto;
+@media (min-width: @tablet) {
+padding-left: 40px;
+padding-right: 40px;
+}
+}
+.center--header {
```
Comment (PetrischakMaxim) [inline source/less/common/center.less:10]: у центровщика обязательно должна быть максимальная ширина и margin: 0 auto;

File: source/less/common/colors.less [hunk 25]
```diff
@@ -1,50 +1,13 @@
...
-@hover-transport-off: #B2B2B2;
-@hover-button-blue: #D1D7E9;
-@pink-required: #F47E7E;
-@pink-background: #FFEFEF;
-@orange-border-catalog: #F3C72B;
-
-@background-color-grey: #EDEFF6;
-@background-color-add-plans: #D4D9EB;
-
+@base-blue: #2D303A;
```
Comment (PetrischakMaxim) [inline source/less/common/colors.less:1]: Именовать перемменные с привязкой к цвету не очень хорошая практика, ведь цвет могут попросить поменять, тогда нужно менять будет и имя переменной. 
Но пока что оставь, с этим еще поработаем.

File: source/js/script.js (hunk 17 above)
Comment (PetrischakMaxim) [reply source/js/script.js]: page__body--modal-visible , наверное лучше будет как-то так

**Review by PetrischakMaxim — COMMENTED** [review]
.page-header задай z-index: 11
.page-header__navigation-list нужна тень box-shadow: 0px -40px 20px rgb(0 0 0 / 40%);
}

и top 0
.page-header__logo z-index 1 и .page-header__navigation-button тоже самое

https://prnt.sc/1s7a0vn попробуй вот так сделать положку
и будет вот так https://prnt.sc/1s7auhv

File: source/index.html [hunk 26]
```diff
@@ -1,128 +1,332 @@
 <!DOCTYPE html>
-<html lang="en">
+<html lang="en" class="page">
```
Comment (PetrischakMaxim) [inline source/index.html:2]: Рекомендую проверять верстку каждый раз в валидаторе

Comment (PetrischakMaxim) [reply source/index.html:2]: Также можно проверить синтаксис БЭМ https://yoksel.github.io/html-tree/

File: source/index.html [hunk 27]
```diff
@@ -1,128 +1,332 @@
...
+
+    <section class="page-main__portfolio portfolio" id="portfolio">
+      <div class="center">
+        <h2 class="portfolio__title header">Portfolio</h2>
+        <p class="portfolio__paragraph text">Duis mollis, est non commodo luctus, nisi erat porttitor ligula,
+          eget lacinia odio sem nec elit.</p>
+        <nav class="portfolio__navigation">
+          <ul class="portfolio__navigation-list">
+            <li class="portfolio__navigation-item"><button class="portfolio__item-button"
+                data-item="all-button">All</button></li>
```
Comment (PetrischakMaxim) [inline source/index.html:165]: можно проще именовать 1 словом, тут например all и тд.
Но уже оставь так)

File: source/index.html [hunk 28]
```diff
@@ -1,128 +1,332 @@
...
+    </div>
   </header>
   <main class="page-main">
-    <section class="page-main__services services">
-      <div class="page-main__slider slider">
-        <button class="slider__button-left"></button>
-        <button class="slider__button-rigth"></button>
-        <img src="" alt="" class="slider__slide">
+    <div class="page-main__slider-wrapper">
+      <div class="center--slider">
```
Comment (PetrischakMaxim) [inline source/index.html:50]: center center--slider

File: source/index.html [hunk 29]
```diff
@@ -1,128 +1,332 @@
...
+                </svg>
+              </a>
+            </li>
+          </ul>
+        </nav>
+      </div>
+    </div>
   </footer>
   <script src="./js/script.min.js"></script>
+  <script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.10.1/jquery.min.js"></script>
```
Comment (PetrischakMaxim) [inline source/index.html:255]: Нее)) Это давай уберем.
Вот [реализация ](https://webdesign.tutsplus.com/tutorials/smooth-scrolling-vanilla-javascript--cms-35165)на чистом JS

File: source/less/blocks/page-header.less [hunk 30]
```diff
@@ -0,0 +1,151 @@
+.page-header {
```
Comment (PetrischakMaxim) [inline source/less/blocks/page-header.less:1]: Проверь пожалуйста стили через  css linter , или командой npm run test.
И надо будет созвониться, покажу как настроить автофикс ошибок и выравнивание кода.

File: source/js/script.js [hunk 31]
```diff
@@ -1 +1,86 @@
...
+      }
+    } else {
+      element.classList.remove(HIDDEN_STATE);
+    }
+  }
+  )
+}
+
+const showGroupOfItemsOnClick = (button, dataItem) => {
+  button.addEventListener('click', () => {
```
Comment (PetrischakMaxim) [inline source/js/script.js:73]: В целом реализация норм) Можно было б еще сделать так.
По клику на кнопку, считывать ее дата атрибут. 
И потом искать елементы с таким же атрибутом и их показывать, а остальные скрывать.

**Review by PetrischakMaxim — COMMENTED** [review]
1. Проверить HTML в валидаторе
2. Отформатировать LESS и устранить ошибки линтера
3. Перестало работать мобильное меню.
4. Burger-menu-icon.svg переименовать с маленькой буквы.
